src/hashset.c

```diff
@@ -32,6 +32,33 @@
  * Functions and types are prefixed by "hashset", macros by "HASHSET". Internal
  * names don't use the prefix. Internal functions are 'static'.
  *
+ * Caution: directly storing non-pointer elements
+ * ----------------------------------------------
+ *
```

Comment:
Some intro about what this is about could be useful:

```suggestion
 * Storing non-pointer values, such as pointer-sized integers (long), in a
 * hashset is possible. Be careful when casting to and from `void *` though.
 *
```